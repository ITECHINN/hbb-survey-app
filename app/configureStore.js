import devTools from 'remote-redux-devtools';
import { createStore, applyMiddleware, compose } from 'redux';
import { persistStore } from 'redux-persist';
import reducer from './redux/reducers';

import createSagaMiddleware from 'redux-saga'
import surveySaga from "./redux/sagas/surveySaga";

export default function configureStore():any {
  const sagaMiddleware = createSagaMiddleware();

  const enhancer = compose(
    applyMiddleware(sagaMiddleware),
    devTools({
      name: 'nativestarterkit', realtime: true,
    }),
  );

  const store = createStore(reducer, enhancer);
  sagaMiddleware.run(surveySaga);

  return store;
}
