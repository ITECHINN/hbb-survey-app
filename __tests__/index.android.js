import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';

import Home from '../app/containers/home';
import configureStore from '../app/configureStore';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const store = configureStore();

it('renders correctly', () => {

  const component = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>
  );

  expect(component).toMatchSnapshot();
});
