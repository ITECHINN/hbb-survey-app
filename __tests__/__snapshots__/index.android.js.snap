// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1314,
      },
      Object {
        "backgroundColor": "#FBFAFA",
      },
    ]
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#3F51B5",
          "borderBottomColor": "#3F51B5",
          "borderBottomWidth": 0,
          "elevation": 3,
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 0,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 0.5,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeBackgroundAndroid={
            Object {
              "borderless": undefined,
              "color": 1308622847,
              "type": "RippleAndroid",
            }
          }
          onLayout={undefined}
          onPress={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 2,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          transparent={true}
          virtual={undefined}
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 28,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 24,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 2,
                    "paddingTop": undefined,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "Roboto_medium",
              "fontSize": 19,
              "fontWeight": undefined,
              "textAlign": "center",
            }
          }
          virtual={undefined}
        >
          Ana Sayfa
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
        virtual={undefined}
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Object {
            "marginLeft": "5%",
            "marginTop": 20,
            "width": "90%",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "wrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 12,
                "paddingLeft": 17,
                "paddingRight": 17,
                "paddingTop": 12,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
              virtual={undefined}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  size={30}
                  style={
                    Object {
                      "flex": 30,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#000",
                            "fontSize": 28,
                          },
                          Object {
                            "color": "#000",
                            "fontSize": 28,
                          },
                          Object {
                            "fontSize": 32,
                          },
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                    virtual={undefined}
                  >
                    
                  </Text>
                </View>
                <View
                  size={70}
                  style={
                    Object {
                      "flex": 70,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                        },
                        Object {
                          "fontSize": 18,
                        },
                      ]
                    }
                    uppercase={false}
                    virtual={undefined}
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  size={30}
                  style={
                    Object {
                      "flex": 30,
                      "flexDirection": "column",
                    }
                  }
                />
                <View
                  size={70}
                  style={
                    Object {
                      "flex": 70,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "Roboto",
                        "fontSize": 16,
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    Aktif anket yoktur
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "marginTop": 18,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#fff",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#ccc",
                    "borderLeftWidth": 0.5,
                    "borderRadius": 2,
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                    "elevation": 3,
                    "flex": 1,
                    "flexWrap": "wrap",
                    "marginBottom": 5,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 5,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 1.5,
                  },
                  Object {
                    "marginLeft": "5%",
                    "width": "90%",
                  },
                ]
              }
              virtual={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "paddingBottom": 12,
                      "paddingLeft": 17,
                      "paddingRight": 17,
                      "paddingTop": 12,
                    },
                    Object {
                      "backgroundColor": "#eee",
                      "paddingBottom": 5,
                      "paddingTop": 5,
                    },
                  ]
                }
                virtual={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 26,
                          "width": 33,
                        },
                        Object {
                          "color": "black",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "Roboto",
                        "fontSize": 16,
                      },
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Son Aktif Anket
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderRadius": 2,
                    "flexDirection": "row",
                    "paddingBottom": 12,
                    "paddingLeft": 17,
                    "paddingRight": 17,
                    "paddingTop": 12,
                  }
                }
                virtual={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "Roboto",
                      "fontSize": 16,
                    }
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Aktif anket bulunamadı
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0,
                "flexDirection": "column",
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              nativeBackgroundAndroid={
                Object {
                  "borderless": undefined,
                  "color": 1308622847,
                  "type": "RippleAndroid",
                }
              }
              onLayout={undefined}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#3F51B5",
                  "borderBottomWidth": null,
                  "borderColor": "#3F51B5",
                  "borderLeftWidth": null,
                  "borderRadius": 2,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "width": "100%",
                }
              }
              testID={undefined}
              virtual={undefined}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  size={20}
                  style={
                    Object {
                      "flex": 20,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#000",
                            "fontSize": 28,
                          },
                          Object {
                            "color": "#fff",
                            "fontSize": 24,
                            "marginLeft": 16,
                            "marginRight": 16,
                            "paddingTop": undefined,
                          },
                          Object {
                            "color": "white",
                          },
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                    virtual={undefined}
                  >
                    
                  </Text>
                </View>
                <View
                  size={80}
                  style={
                    Object {
                      "flex": 80,
                      "flexDirection": "column",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "Roboto_medium",
                        "fontSize": 14,
                        "lineHeight": 19,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    Anket Listesi
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0,
                "flexDirection": "column",
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              nativeBackgroundAndroid={
                Object {
                  "borderless": undefined,
                  "color": 1308622847,
                  "type": "RippleAndroid",
                }
              }
              onLayout={undefined}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#3F51B5",
                  "borderBottomWidth": null,
                  "borderColor": "#3F51B5",
                  "borderLeftWidth": null,
                  "borderRadius": 2,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "width": "100%",
                }
              }
              testID={undefined}
              virtual={undefined}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  size={20}
                  style={
                    Object {
                      "flex": 20,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#000",
                            "fontSize": 28,
                          },
                          Object {
                            "color": "#fff",
                            "fontSize": 24,
                            "marginLeft": 16,
                            "marginRight": 16,
                            "paddingTop": undefined,
                          },
                          Object {
                            "color": "white",
                          },
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                    virtual={undefined}
                  >
                    
                  </Text>
                </View>
                <View
                  size={80}
                  style={
                    Object {
                      "flex": 80,
                      "flexDirection": "column",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "Roboto_medium",
                        "fontSize": 14,
                        "lineHeight": 19,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    Son Tamamlanmış Anketler
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
